### ISSUE TITLE
Add test to compare public API between Microsoft-built and source-built SDKs

### ISSUE THREAD

**Issue by mthalman** [dotnet/source-build#4795]
A recent issue occurred in .NET 9 where the Microsoft.AspNetCore.Server.HttpSys assembly was essentially empty in the source-built SDK: https://github.com/dotnet/aspnetcore/issues/59166. This was not caught prior to GA because there were no tests that validated the usage of this assembly on a source-built SDK.

We should consider defining a test, similar to the exists tests in [SdkContentTests.cs](https://github.com/dotnet/sdk/blob/main/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.Tests/SdkContentTests.cs), that compares the public API between the Microsoft-built and source-built SDKs. The assertion is that the APIs are identical between the two. As a general purpose test, this would apply to all assemblies and would have caught this issue well in advance of GA release.

**Comment by MichaelSimons** [comment]
[Triage] It is likely that a baseline mechanism is going to be needed as there are components intentionally excluded from SB.  Exclusions would likely be needed at api, class, and namespace levels to help with maintainability.  There could also be some difference caused by version differences between the two builds.

**Comment by ViktorHofer** [comment]
Yes, I see APICompat as the right tool to do this. With strict mode enabled. I would probably use the msbuild entry point by adding a proj file that invokes the `ValidateAssembliesTask`.
